webrender/src/device/gfx/device.rs

```diff
@@ -3434,15 +3444,15 @@ impl<B: hal::Backend> Device<B> {
         unsafe {
             match self.swap_chain.as_mut() {
                 Some(swap_chain) => {
-                    let submission = Submission {
+                    let submission = hal::queue::Submission {
                         command_buffers: self.command_pool[self.next_id].command_buffers(),
                         wait_semaphores: Some((
                             &self.image_available_semaphore,
                             PipelineStage::BOTTOM_OF_PIPE,
                         )),
                         signal_semaphores: Some(&self.render_finished_semaphore),
                     };
-                    self.queue_group.queues[0]
+                    self.queue_group.queues[0].as_raw_mut()
```

Comment:
I think we should not call `as_raw_mut()` here since it's still related to the typed `CommandQueue<B>`

Reply:
The problem with Queues is that some parts of the API returns or expects them, for example [struct Gpu](https://docs.rs/gfx-hal/0.2.1/gfx_hal/struct.Gpu.html) or the swapchain's [present](https://docs.rs/gfx-hal/0.2.1/gfx_hal/window/trait.Swapchain.html#method.present) method. Both cases would require some conversion if we'd like to do this on our side.

Reply:
You can acquire a `B::CommandQueue` by calling `take_raw` instead of `take` here : https://github.com/szeged/webrender/pull/301/files#diff-cdf03c96bd34cb98c2d09b0be7658232R308, and instead of swapchain's present you can use `B::CommandQueue`'s present: https://github.com/gfx-rs/gfx/blob/master/src/hal/src/queue/mod.rs#L78 and the submit function as well.

Reply:
Ah ok, that could work.